web/src/engine/websites/Sukima.ts

```diff
@@ -0,0 +1,191 @@
+        for (const book of books.contents) {
+            for (const chapter of book.stories) {
+                const id = `/bv/t/${chapter.title_code}/v/${chapter.volume}/s/${chapter.story}/p/1`;
+                const title = `(${chapter.volume.toString().padStart(3, '0')}-${chapter.story.toString().padStart(3, '0')})${chapter.info.text.trim()}`.replace(manga.Title, '').trim();
+                chapters.push(new Chapter(this, manga, id, title));
+            }
+        }
+        return chapters;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const request = new FetchRequest(new URL(chapter.Identifier, this.URI).href);
+        const pages = await FetchWindowScript<PAGE_INFO[]>(request, pageScript, 1500);
+        return pages.map(page => new Page(this, chapter, new URL(page.page_url), { Referer: request.url, payload: JSON.stringify(page) }));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const blob = await Common.FetchImageAjax.call(this, page, priority, signal);
+        return this.descrambleImage(blob, JSON.parse(page.Parameters.payload as string) as PAGE_INFO);
+    }
+
+    async descrambleImage(blob: Blob, payload: PAGE_INFO): Promise<Blob> {
+        const settings = HakuNeko.SettingsManager.OpenScope();
+        const format = settings.Get<Text>(GlobalKey.DescramblingFormat).Value;
+        const quality = settings.Get<Numeric>(GlobalKey.DescramblingQuality).Value;
+
+        const bitmap = await createImageBitmap(blob);
+        return new Promise(resolve => {
+
+            const canvas = document.createElement('canvas');
+            canvas.width = bitmap.width;
+            canvas.height = bitmap.height;
+            const ctx = canvas.getContext('2d');
+
+            const xSplitCount = Math.floor(bitmap.width / payload.blocklen);
+            const ySplitCount = Math.floor(bitmap.height / payload.blocklen);
+            let count = 0;
+            ctx.drawImage(bitmap, 0, ySplitCount * payload.blocklen + 1, bitmap.width, bitmap.height - ySplitCount * payload.blocklen, 0, ySplitCount * payload.blocklen, bitmap.width, bitmap.height - ySplitCount * payload.blocklen);
+
+            for (let col = 0; col < xSplitCount; col++) {
+                for (let row = 0; row < ySplitCount; row++) {
+                    const dx = payload.shuffle_map[count][0];
+                    const dy = payload.shuffle_map[count][1];
+                    const sx = col * payload.blocklen;
+                    const sy = row * payload.blocklen;
+                    ctx.drawImage(bitmap, sx, sy, payload.blocklen, payload.blocklen, dx, dy, payload.blocklen, payload.blocklen);
+                    count++;
+                }
+            }
+
+            canvas.toBlob(data => {
+                resolve(data);
+            }, format, quality / 100);
+        });
+    }
+
+    async fetchPOST<T>(uri : string, body : string) {
```

Comment:
Suggestion to make `body` parameter of type `unknown` or `Object` and perform the JSON serialisation within this method